src/kernel/mempool_persist.cpp

```diff
@@ -57,9 +59,19 @@ bool LoadMempool(CTxMemPool& pool, const fs::path& load_path, Chainstate& active
     try {
         uint64_t version;
         file >> version;
-        if (version != MEMPOOL_DUMP_VERSION) {
+        if (version == 1) {
+            // No XOR-key present
+        } else if (version == MEMPOOL_DUMP_VERSION) {
+            file >> xor_key;
+        } else {
             return false;
         }
+        const auto pos{std::ftell(file.Get())};
+        file.fclose();
+        CAutoFile file{xor_open()};
```

Comment:
In eeee4a5779ea20d859cff2e411ad46dd52384f1d "mempool: persist with XOR"

Closing and reopening the file once we have the xor seems kinda clunky. Why not have a function on `CAutoFile` that lets us set the xor key after the fact?